client/src/app/shared/shared-video-miniature/video-actions-dropdown.component.ts

```diff
@@ -180,10 +185,11 @@ export class VideoActionsDropdownComponent implements OnChanges {
   }
 
   isVideoDownloadable () {
-    return this.video &&
-      this.video.isLive !== true &&
-      this.video instanceof VideoDetails &&
-      this.video.downloadEnabled
+    if (!this.video || this.video.isLive) {
```

Comment:
I'd prefer to keep the code like it was, and add a condition where:
 * It's a video details object and so we check `downloadEnabled`
 * It's not then check the user is logged in and owns the video or has the `SEE_ALL_VIDEOS` (you can use and rename `canSeeStats` method of `video.model.ts` if you want)